packages/block-library/src/index.native.js

```diff
@@ -182,6 +182,78 @@ addFilter(
 	}
 );
 
+const ineligibleTransformations = {
+	'core/paragraph': [ 'core/group', 'core/columns' ],
+};
+const supportedTransformations = [ 'core/paragraph' ];
+
+const processedIneligibleTransformations = map(
+	ineligibleTransformations,
+	( toBlocks, fromBlock ) =>
+		toBlocks.map( ( toBlock ) => ( { to: toBlock, from: fromBlock } ) )
+).flatMap( ( el ) => el );
+
+/**
+ * Limits transformations to:
+ *   1. Blocks that support transformations
+ *   2. Blocks that aren't in the ineligibleTransformations set
+ *   3. Can only transform to blocks that are supported
+ *     TODO: this should be taken care of by whitelist, no?
```

Comment:
@chipsnyder This is the specific code relevant to the question I asked in the description: whether or not explicitly checking for unsupported blocks is necessary when there's a whitelist.

Reply:
I love that we call out the goals here in the description. 

I agree using the allow list on the `to` and `from` sides seems to meet the goals of `#1` and `#3`.

I tried various cases to see If I could get a weird scenario, and it all seemed to work nicely. While testing, I noticed that if I add support for `core/group` or `core/columns`, they would be enabled for the heading block. I'll leave that up to @iamthomasbishop if we want to support that or not. but it might be worth adding. `'core/heading': [ 'core/group', 'core/columns' ]` to `ineligibleTransformations` WDYT?

----- 

I had a different question/suggestion related to this. Since we have the `transform.js` files on each block, I wonder if we could utilize a field in there or one added to a possible `transform.native.js` to track the `ineligibleTransformations`. That way, the disallow list is more closely tied to how we handle the allow list. That said, I didn't mess around with that to see what hardships that might introduce. WDYT?

Reply:
I did run into one scenario that I think we might want to at least consider. If I add support for `core/group` or `core/columns` then I'll see transform options on Quote (which isn't in the allowed list of `supportedTransformations`). 

I don't know if this has to be fixed now but it might be good to consider a solution and/or leave a note that it's something to be mindful of as we add more transformations we'll need to check all of the blocks it transforms to. WDYT?

Reply:
Hrmm, I think that it is something that needs to be considered. @iamthomasbishop what's the desired behaviour here? If we add in a supported transformation, do we want other (unsupported) blocks to be able to transform to the supported block?

Reply:
@chipsnyder  For the above point: I 100% agree with moving the lists out of this file to jive better with the `transform.js` file; I separated it out for that reason but I honestly wasn't sure where to put it. If I add a `transform.native.js` file, which block would I add it under? Each block has its own `transform.js` file atm so I wasn't sure how to consolidate that.

Reply:
I think If I were to describe the goal, I would say: "I don't want paragraph blocks to transform into a group or column blocks" So worded that way, I would probably put an array in the paragraph's `transform.js` or `transform.native.js` file. However, that may not be a great way to make "architectural" decisions 😄. 

As for where it lives in those definitions, I would probably add something at the same level as the to/from definitions.

So in a `transform.native.js` you might have something like
```
const transforms = {
	from: [ ... ],
        to: [ ... ],
	ineligibleTransformations: [ ... ] 
};
```

@gziolo might have an opinion or guidance to add too :)

Reply:
Can you explain what is the goal of this filter? I don't have enough context to answer your question. First, I would have to understand why do you have to apply any change to the list of transformations. Assuming you support all those blocks, why would it be different in the mobile app?

Reply:
@gziolo I think the idea is that it's a lot harder to do certain things on mobile UI, for example, selecting multiple blocks in order to convert to columns/groups.

Reply:
You can still want to convert (in practice wrap) a single block with a group or column. Why would you prevent it?

Reply:
Some of the reasons I could see would be to use it as a mechanism to turn off a transformation that for one reason or another is leading to a problematic experience on mobile. Used as a short term safety net. 

@iamthomasbishop you had mentioned wanting to avoid these transformations for v1. I'm not sure if you had any more thoughts there that you would want to add.

Reply:
I can think only about the case when the block you are targeting with transforms is unsupported in the mobile app. Then it doesn't matter how you tackle it in the filter since it's a temp solution. I would only avoid changing the original definition of the block that web uses.